### PR TITLE
typing(ui): add `slice` to `ActionRow.__getitem__` and `__delitem__` annotations

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Fixes type-checking for expressions like `row[1:4]` and `del row[2:3]`.
Also adds `ActionRow.__iter__` since it was missing before (though it worked implicitly through `__getitem__`), and to workaround a pyright bug.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by EQUENOS — APPROVED** [review]

**Comment by Chromosomologist** [comment]
Ah, good idea, actually; I forgot about slices when I did the refactor. Anyways, LGTM!

File: disnake/ui/action_row.py [hunk 1]
```diff
@@ -89,7 +90,15 @@ class ActionRow(Generic[UIComponentT]):
...
-            Returns the component at position ``i``.
+            Returns the component at position ``i``. Also supports slices.
+
+        .. describe:: len(x)
+
+            Returns the number of components in this row.
+
+        .. describe:: iter(x)
+
+            Returns an iterator for the components in this row.
```
Comment (onerandomusername) [inline disnake/ui/action_row.py:101]: missing versionadded indicators.

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: disnake/ui/action_row.py (hunk 1 above)
Comment (shiftinv) [reply disnake/ui/action_row.py]: https://github.com/DisnakeDev/disnake/pull/624/commits/aba82f40a1b97986b8ba0351abea3d971f165435

**Review by onerandomusername — APPROVED** [review]
looks good, thanks

**Pull request merged**